File: client/src/models/campaign/encounter/Encounter.ts
import Party from "../Party";
import {MinionGroup} from "../../actor/npc/Minion";
import Rival from "../../actor/npc/Rival";
import Nemesis from "../../actor/npc/Nemesis";
import InitiativeSlot from "./InitiativeSlot";

export default interface Encounter {
    type: Type
    party: Party
    minions: MinionGroup[]
    rivals: Rival[]
    nemeses: Nemesis[]
    slots: InitiativeSlot[]
}

export enum Type {
    Combat = 'Combat',
    Social = 'Social',
}

File: server/src/main/java/com/github/genraven/genesys/domain/campaign/encounter/Encounter.java
package com.github.genraven.genesys.domain.campaign.encounter;

import com.fasterxml.jackson.annotation.JsonValue;
import com.github.genraven.genesys.domain.actor.npc.MinionGroup;
import com.github.genraven.genesys.domain.actor.npc.Nemesis;
import com.github.genraven.genesys.domain.actor.npc.Rival;
import com.github.genraven.genesys.domain.campaign.Party;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;

import java.util.ArrayList;
import java.util.List;

@Data
public class Encounter {

    private Type type = Type.COMBAT;
    private Party party = new Party();
    private List<Rival> enemyRivals = new ArrayList<>();
    private List<Nemesis> enemyNemeses = new ArrayList<>();
    private List<MinionGroup> enemyMinionGroups = new ArrayList<>();
    private List<InitiativeSlot> slot = new ArrayList<>();

    @Getter
    @AllArgsConstructor
    public enum Type {
        COMBAT("Combat"),
        SOCIAL("Social");

        @JsonValue
        private final String label;
    }
}
